src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala

```diff
@@ -174,16 +174,51 @@ class MesosTaskBuilder @Inject()(val conf: SchedulerConfiguration) {
       v.mode.map { m =>
         volumeBuilder.setMode(Volume.Mode.valueOf(m.toString.toUpperCase))
       }
+      v.external.foreach { e =>
+        volumeBuilder.setSource(Volume.Source.newBuilder()
+            .setType(Volume.Source.Type.DOCKER_VOLUME)
+            .setDockerVolume(Volume.Source.DockerVolume.newBuilder()
+              .setDriver(e.provider)
+              .setName(e.name)
+              .setDriverOptions(Parameters.newBuilder()
+                .addAllParameter(e.options.map(_.toProto()).asJava).build()
+              ).build()
+            ).build()
+        ).build()
+      }
 
       volumeBuilder.build()
     }.foreach(builder.addVolumes)
-    builder.setType(ContainerInfo.Type.DOCKER)
-    builder.setDocker(DockerInfo.newBuilder()
-      .setImage(job.container.image)
-      .setNetwork(DockerInfo.Network.valueOf(job.container.network.toString.toUpperCase))
-      .setForcePullImage(job.container.forcePullImage)
-      .addAllParameters(job.container.parameters.map(_.toProto).asJava)
-      .build()).build
+
+    job.container.`type` match {
+      case ContainerType.DOCKER =>
+        builder.setType(ContainerInfo.Type.DOCKER)
+        builder.setDocker(DockerInfo.newBuilder()
+          .setImage(job.container.image)
+          .setNetwork(DockerInfo.Network.valueOf(job.container.network.toString.toUpperCase))
+          .setForcePullImage(job.container.forcePullImage)
+          .addAllParameters(job.container.parameters.map(_.toProto()).asJava)
+          .build())
+      case ContainerType.MESOS =>
+        builder.setType(ContainerInfo.Type.MESOS)
+        builder.setMesos(ContainerInfo.MesosInfo.newBuilder()
+          .setImage(Image.newBuilder()
+            // TODO add APPC image support
+            .setType(Image.Type.DOCKER)
+            .setDocker(Image.Docker.newBuilder()
+              .setName(job.container.image)
+              // TODO add setCredential
+              .build())
+            .setCached(!job.container.forcePullImage)
+            .build())
+          .build())
+    }
+    job.container.networkName.foreach {
+      n => builder.addNetworkInfos(NetworkInfo.newBuilder()
```

Comment:
Agrees this is not sufficient, I keep this minimal implementation for now, because I (and maybe others) have many deployed jobs using this parameter, we'll add a more complete networkinfo support in the future.

`name` is the only required param for CNI networks, other params, like `protocol`, `labels` or `port_mappings` are optional.

The marathon's ip-per-task spec does not support multiple networks.
IMO, we should use a spec closer to the networkInfo proto https://github.com/apache/mesos/blob/81cd023eb9945a22c220edc966393dcfcdbce256/include/mesos/mesos.proto#L1922